src/librustc_mir/build/expr/as_rvalue.rs

```diff
@@ -181,6 +181,9 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
             ExprKind::Adt {
                 adt_def, variant_index, substs, fields, base
             } => { // see (*) above
+                let is_union = adt_def.adt_kind() == ty::AdtKind::Union;
+                let active_field_index = if is_union { Some(fields[0].name.index()) } else { None };
+
```

Comment:
>alternatively, an Union AggregateKind could be added

IIRC, I tried this and didn't like the result (too much common boilerplate code for `Adt` and `Union`)